### PR TITLE
Add spark warehouse to test

### PR DESCRIPTION
**Pull request by JCZuurmond**
To start development on the Spark dialect. This PR adds the spark warehouse to the tests.

### CODE REVIEW

File: tests/warehouses/spark_fixture.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from typing import List
+
+from tests.common.warehouse_fixture import WarehouseFixture
+
+
+class SparkFixture(WarehouseFixture):
```
Comment (JCZuurmond) [inline tests/warehouses/spark_fixture.py:6]: Which methods does this fixture need?

File: tests/warehouses/spark_suite.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from tests.common.sql_test_case import TARGET_SPARK
+from tests.common.sql_test_suite import SqlTestSuite
+
+
+class HiveSuite(SqlTestSuite):
+
+    def setUp(self) -> None:
+        self.target = TARGET_SPARK
```
Comment (JCZuurmond) [inline tests/warehouses/spark_suite.py:8]: Maybe could be a class attribute. I copy pasted this from another suite.

File: tests/warehouses/spark_cfg.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+type: spark
```
Comment (JCZuurmond) [inline tests/warehouses/spark_cfg.yml:1]: I put in some basics for now. We could add multiple `methods` and some other parameters to connect to different locations. See [`dbt-spark`](https://github.com/fishtown-analytics/dbt-spark#configuring-your-profile) as example.

Comment (vijaykiran) [reply tests/warehouses/spark_cfg.yml:1]: I think this is good start - we can add additional parameters to differentiate later

File: tests/warehouses/spark_suite.py (hunk 2 above)
Comment (vijaykiran) [reply tests/warehouses/spark_suite.py:8]: Good point, Will add to my refactor list.

**Pull request merged**